nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/RemoveItemFromListPropertyTest.java

```diff
@@ -115,60 +117,82 @@ public void clearRepo() {
     }
 
     @Test
     @SuppressWarnings("unchecked")
     public void removeFirstItemFromListPropertyTest() throws Exception {
-        //remove the first item
-        DocumentModel resultDoc = removeItemsFromListProperty(0);
+        // remove the first item
+        DocumentModel resultDoc = removeItemsFromListProperty("ds:fields", 0);
 
         List<Map<String, String>> dbFields = (List<Map<String, String>>) resultDoc.getPropertyValue("ds:fields");
         assertEquals(1, dbFields.size());
 
         Map<String, String> properties = dbFields.get(0);
-        assertEquals(properties.get("fieldType"), "unicTypeAdded2");
+        assertEquals("unicTypeAdded2", properties.get("fieldType"));
     }
 
     @Test
     @SuppressWarnings("unchecked")
     public void removeLastItemFromListPropertyTest() throws Exception {
         // Remove the last item
-        DocumentModel resultDoc = removeItemsFromListProperty(1);
+        DocumentModel resultDoc = removeItemsFromListProperty("ds:fields", 1);
         List<Map<String, String>> dbFields = (List<Map<String, String>>) resultDoc.getPropertyValue("ds:fields");
         assertEquals(1, dbFields.size());
 
         Map<String, String> properties = dbFields.get(0);
-        assertEquals(properties.get("fieldType"), "unicTypeAdded");
+        assertEquals("unicTypeAdded", properties.get("fieldType"));
     }
 
     @Test(expected = OperationException.class)
     @SuppressWarnings("unchecked")
     public void removeNonExistentItemFromListPropertyTest() throws Exception {
         // Not possible to remove the index:2, because the list only have two items
-        DocumentModel resultDoc = removeItemsFromListProperty(2);
+        DocumentModel resultDoc = removeItemsFromListProperty("ds:fields", 2);
         List<Map<String, String>> dbFields = (List<Map<String, String>>) resultDoc.getPropertyValue("ds:fields");
         assertEquals(1, dbFields.size());
 
         Map<String, String> properties = dbFields.get(0);
-        assertEquals(properties.get("fieldType"), "unicTypeAdded");
+        assertEquals("unicTypeAdded", properties.get("fieldType"));
     }
 
-    protected DocumentModel removeItemsFromListProperty(Integer index) throws OperationException {
-        // Remove all the fields
+    protected DocumentModel removeItemsFromListProperty(String xpath, Integer index) throws OperationException {
         OperationContext ctx = new OperationContext(coreSession);
         ctx.setInput(doc);
-        OperationChain chain = new OperationChain("testRemoveItemFromPropertyChain");
 
-        String xpath = "ds:fields";
-        OperationParameters parameters = chain.add(RemoveItemFromListProperty.ID).set("xpath", xpath);
-
-        if (index != null) parameters.set("index", index);
+        Map<String, Object> params = new HashMap<>();
+        params.put("xpath", xpath);
+        params.put("index", index);
+        return (DocumentModel) service.run(ctx, RemoveItemFromListProperty.ID, params);
```

Comment:
What do you think about:
```java
Map<String, Object> params = Map.of("xpath", xpath, "index", index);
```

Reply:
We should refrain from using Java 11 features for anything that could potentially be backported...

Reply:
Isn't it a shame not to use new features on master. Being careful while backporting should be enough no?

Reply:
As you wish, I'm just warning about potential headaches when backporting. But it's true this shouldn't prevent us from cleaning up code.